src/components/data/data-utils.js

```diff
@@ -695,6 +696,76 @@ export const ModalToolButton = ({ icon, text, disabled, ...props }) => {
   ])
 }
 
+export const DeleteEntityColumnModal = ({ workspaceId: { namespace, name }, column: { entityType, attributeName }, onDismiss, onSuccess }) => {
+  const [deleting, setDeleting] = useState(false)
+  const [deleteConfirmation, setDeleteConfirmation] = useState('')
+
+  const signal = Utils.useCancellation()
+
+  const deleteColumn = async () => {
+    try {
+      setDeleting(true)
+      await Ajax(signal).Workspaces.workspace(namespace, name).deleteEntityColumn(entityType, attributeName)
+      onDismiss()
+      onSuccess()
+    } catch (e) {
+      reportError('Unable to modify column', e)
+      setDeleting(false)
+    }
+  }
+
+  return h(Modal, {
+    title: 'Delete Column',
+    onDismiss,
+    okButton: h(ButtonPrimary, {
+      onClick: deleteColumn,
+      disabled: _.toLower(deleteConfirmation) !== 'delete column',
+      tooltip: _.toLower(deleteConfirmation) !== 'delete column' ? 'You must type the confirmation message' : undefined
+    }, 'Delete column')
+  }, [
+    div(['Are you sure you want to permanently delete the column ',
+      span({ style: { fontWeight: 600, wordBreak: 'break-word' } }, attributeName),
+      '?']),
+    div({
+      style: { fontWeight: 500, marginTop: '1rem' }
+    }, 'This cannot be undone.'),
+    div({ style: { marginTop: '1rem' } }, [
+      label({ htmlFor: 'delete-column-confirmation' }, ['Please type "Delete Column" to continue:']),
```

Comment:
my personal $.02 is that we should ask the user to type the column name, instead of a generic "Delete Column". Since the column name is included in the message above, they can always copy/paste it. If our UX team feels strongly about using the generic "Delete Column" then so be it, I won't push this

Reply:
oh, I know … I don't like that one either :)